app/components/bali/calendar/component.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+module Bali
+  module Calendar
+    class Component < ApplicationViewComponent
+      attr_reader :start_date, :period, :start_attribute, :end_attribute, :events,
+                  :template, :all_week, :show_date
+
+      renders_one :footer
+      renders_one :header, Header::Component
```

Comment:
Aqui podemos incluirle el @start_attribute al Header::Component desde el Calendar::Component para que el usuario del componente no lo tenga que especificar en el header tambien.

```suggestion
      renders_one :header, ->(start_date:, period: :month, route_name: nil, period_switch: true, **options) do
        Header::Component.new(
          start_date: start_date,
          period: period,
          route_name: route_name,
          period_switch: period_switch,
          start_attribute: @start_attribute,
          **options
      end
```

Reply:
https://viewcomponent.org/guide/slots.html#lambda-slots

"Lambda slots are able to access state from the parent ViewComponent:"